index.js

```diff
@@ -4,9 +4,11 @@ import inherits from 'inherits'
  * Create a custom constructor which can be modified without affecting the
  * original class.
  *
- * @template {Function} Class
+ * @template Instance
+ * @template Args
+ * @template {{ new(...Args): Instance }} Class
  * @param {Class} Super
- * @returns {Class}
+ * @return {(new(...Args) => Instance) & ((...Args) => Instance)}
```

Comment:
returning class directly appears to run into https://stackoverflow.com/questions/17382143/create-a-new-object-from-type-parameter-in-generic-class
This overcomes this a constructor and factory are declared and use the reflected constructor and instance information